server/src/session_aware_backend_handler.rs

```diff
@@ -0,0 +1,441 @@
+    }
+
+    async fn registration_start(
+        &self,
+        request: lldap_auth::registration::ClientRegistrationStartRequest,
+    ) -> std::result::Result<
+        lldap_auth::registration::ServerRegistrationStartResponse,
+        lldap_domain_model::error::DomainError,
+    > {
+        self.inner.registration_start(request).await
+    }
+
+    async fn registration_finish(
+        &self,
+        request: lldap_auth::registration::ClientRegistrationFinishRequest,
+    ) -> std::result::Result<(), lldap_domain_model::error::DomainError> {
+        self.inner.registration_finish(request).await
+    }
+}
+
+// Implement ReadSchemaBackendHandler by delegating to the inner handler
+#[async_trait]
+impl ReadSchemaBackendHandler for SessionAwareBackendHandler {
+    async fn get_schema(&self) -> Result<lldap_domain::schema::Schema> {
+        self.inner.get_schema().await
+    }
+}
+
+// Implement SchemaBackendHandler by delegating to the inner handler
+#[async_trait]
+impl SchemaBackendHandler for SessionAwareBackendHandler {
+    async fn add_user_attribute(&self, request: CreateAttributeRequest) -> Result<()> {
+        self.inner.add_user_attribute(request).await
+    }
+
+    async fn add_group_attribute(&self, request: CreateAttributeRequest) -> Result<()> {
+        self.inner.add_group_attribute(request).await
+    }
+
+    async fn delete_user_attribute(&self, name: &AttributeName) -> Result<()> {
+        self.inner.delete_user_attribute(name).await
+    }
+
+    async fn delete_group_attribute(&self, name: &AttributeName) -> Result<()> {
+        self.inner.delete_group_attribute(name).await
+    }
+
+    async fn add_user_object_class(&self, name: &LdapObjectClass) -> Result<()> {
+        self.inner.add_user_object_class(name).await
+    }
+
+    async fn add_group_object_class(&self, name: &LdapObjectClass) -> Result<()> {
+        self.inner.add_group_object_class(name).await
+    }
+
+    async fn delete_user_object_class(&self, name: &LdapObjectClass) -> Result<()> {
+        self.inner.delete_user_object_class(name).await
+    }
+
+    async fn delete_group_object_class(&self, name: &LdapObjectClass) -> Result<()> {
+        self.inner.delete_group_object_class(name).await
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use lldap_auth::opaque::server::ServerSetup;
+    use lldap_domain::requests::UpdateUserRequest;
+    use lldap_domain::types::UserId;
+    use std::collections::HashSet;
+    use std::sync::{Arc, RwLock};
+    use tokio;
+
+    async fn create_test_backend() -> SqlBackendHandler {
+        let db = sea_orm::Database::connect("sqlite::memory:")
+            .await
+            .expect("Failed to create in-memory database");
+
+        // Create a minimal OPAQUE setup for testing
+        use rand::SeedableRng;
+        let mut rng = rand_chacha::ChaCha20Rng::from_seed([0u8; 32]);
+        let opaque_setup = ServerSetup::new(&mut rng);
+
+        SqlBackendHandler::new(opaque_setup, db)
```

Comment:
Check out the `sql_opaque_handler` tests, to get an initialized db, and a new random private key.